### PR TITLE
add bloom filter metrics

### PR DESCRIPTION
**Pull request by ceyonur**
## Why this should be merged

Bumps AvalancheGo and adds bloom filter metrics. 

Closes: #454 

## How this works

Passes registerer and namespace as input to new bloom filter functions.  

## How this was tested

Existing and AvalancheGo UTs should cover this.

### CODE REVIEW

**Comment by aaronbuchwald** [comment]
Looks like this needs to bump the AvalancheGo version in `scripts/versions.sh` and then this LGTM

**Review by darioush — APPROVED** [review]

File: scripts/tests.e2e.sh [hunk 1]
```diff
@@ -30,15 +33,25 @@ trap cleanup EXIT
...
+# check if given reference is a tag or branch
+if git rev-parse --quiet --verify "origin/${avalanche_version}^{commit}" >/dev/null; then
+  echo "checking out branch ${avalanche_version}"
+  ref="origin/${avalanche_version}"
+else
+  echo "checking out tag ${avalanche_version}"
+  ref="${avalanche_version}"
+fi
+
+git checkout -B "test-${avalanche_version}" "${ref}"
```
Comment (StephenButtolph) [inline scripts/tests.e2e.sh:54]: Does it make sense for us to be targeting a branch for CI? It feels like we should be explicit and target either a tag or a commit imo.

**Review by StephenButtolph — APPROVED** [review]
code lgtm, not sure if we want to keep the script changes

File: scripts/tests.e2e.sh (hunk 1 above)
Comment (ceyonur) [reply scripts/tests.e2e.sh]: reverted it

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**